cpp/src/arrow/compute/api_scalar.h

```diff
@@ -278,6 +278,23 @@ struct ARROW_EXPORT DayOfWeekOptions : public FunctionOptions {
   uint32_t week_start;
 };
 
+struct ARROW_EXPORT WeekOptions : public FunctionOptions {
+ public:
+  explicit WeekOptions(bool week_starts_monday = true, bool count_from_zero = false,
+                       bool first_week_in_year = false);
+  constexpr static char const kTypeName[] = "WeekOptions";
+  static WeekOptions Defaults() { return WeekOptions{}; }
+
+  /// What day does the week start with (Monday=true, Sunday=false)
+  bool week_starts_monday;
```

Comment:
Hmm... did you deliberately choose a different option name and semantics compared with `DayOfWeekOptions::week_start`?

Reply:
Week start is an int and can have values from 1 to 7. Here I'd like to limit to two options - Monday/Sunday hence the boolean and different name.

Reply:
Why would you like to limit it? What is the rationale for that?

Reply:
I've not found an example of Week function that offers options other than Monday or Sunday. So I'd not like to offer more options if possible.

Reply:
I'm a bit curious why it would be useful for DayOfWeek but not for Week. Ideally we should try to use similar idioms when similar concepts are involved.

Reply:
I would actually switch `DayOfWeekOptions::week_start` to `bool DayOfWeekOptions::week_starts_monday` if everyone's up for it. Then we could just use `WeekOptions` for both and it would be just slightly wrong.

Reply:
I think that's reasonable.